patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -4924,21 +5162,25 @@
  				{
  					flag2 = true;
  				}
-@@ -37645,6 +_,14 @@
+@@ -37648,8 +_,18 @@
+ 				{
  					flag2 = true;
  				}
- 				if ((shoot == 145 || shoot == 146 || shoot == 147 || shoot == 148 || shoot == 149) && this.itemAnimation < this.itemAnimationMax - 5)
++				if (!PlayerHooks.ConsumeAmmo(this, sItem, item))
```

Comment:
The patch difference can be made smaller by using the bitwise operator
`flag2 = !PlayerHooks.ConsumeAmmo(this, sItem, item) & !ItemLoader.ConsumeAmmo(sItem, item, this);`

Reply:
Yes